platform/entanglement/src/services/LinkService.js

```diff
@@ -39,42 +39,28 @@ define(
         }
 
         LinkService.prototype.validate = function (object, parentCandidate) {
-            if (!parentCandidate || !parentCandidate.getId) {
-                return false;
-            }
-            if (parentCandidate.getId() === object.getId()) {
-                return false;
-            }
-            if (parentCandidate.getModel().composition.indexOf(object.getId()) !== -1) {
-                return false;
-            }
-            return this.policyService.allow(
-                "composition",
-                parentCandidate.getCapability('type'),
-                object.getCapability('type')
-            );
+            var objectId = object.getId();
```

Comment:
I'll agree to disagree on code style and change it back to what it was. Flow control that happens to behave like logic has some drawbacks but they are tiny, not worth continued debate.

Don't really understand the second point on JSDoc. What's the semantic difference between `linkService.validate` and `moveService.validate`?